c-layer/lib.cxx

```diff
@@ -132,16 +138,39 @@ void free_status(struct Status *status) noexcept {
 
 static zypp::Resolvable::Kind kind_to_zypp_kind(RESOLVABLE_KIND kind) {
   switch (kind) {
-    case RESOLVABLE_PACKAGE: return zypp::Resolvable::Kind::package;
-    case RESOLVABLE_SRCPACKAGE: return zypp::Resolvable::Kind::srcpackage;
-    case RESOLVABLE_PATTERN: return zypp::Resolvable::Kind::pattern;
-    case RESOLVABLE_PRODUCT: return zypp::Resolvable::Kind::product;
-    case RESOLVABLE_PATCH: return zypp::Resolvable::Kind::patch;     
+  case RESOLVABLE_PACKAGE:
+    return zypp::Resolvable::Kind::package;
+  case RESOLVABLE_SRCPACKAGE:
+    return zypp::Resolvable::Kind::srcpackage;
+  case RESOLVABLE_PATTERN:
+    return zypp::Resolvable::Kind::pattern;
+  case RESOLVABLE_PRODUCT:
+    return zypp::Resolvable::Kind::product;
+  case RESOLVABLE_PATCH:
+    return zypp::Resolvable::Kind::patch;
   }
   return zypp::Resolvable::Kind::nokind; // should not happen
 }
 
-void resolvable_select(const char* name, enum RESOLVABLE_KIND kind, struct Status* status) noexcept {
+static zypp::ResStatus::TransactByValue convert_selected(enum RESOLVABLE_SELECTED who) {
+  switch (who) {
+    case RESOLVABLE_SELECTED::SOLVER_SELECTED: return zypp::ResStatus::SOLVER;
+    case RESOLVABLE_SELECTED::APPLICATION_SELECTED: return zypp::ResStatus::APPL_HIGH;
+    case RESOLVABLE_SELECTED::USER_SELECTED: return zypp::ResStatus::USER;
+    case RESOLVABLE_SELECTED::NOT_SELECTED: return zypp::ResStatus::SOLVER; // this should not happen as it should result in early exit
+  }
+
+  // should not happen
```

Comment:
In general, if it should not happen, we should just panic, instead of propagating an "almost correct" value to some other place, causing "fun" debugging sessions.

And BTW, I find the [Rust `expect` message style](https://doc.rust-lang.org/std/result/enum.Result.html#recommended-message-style) helpful: when things go wrong, instead of saying "unexpected value", which is Technically Correct but not helpful to the user or the programmer making the hotfix, the message should say, 
> [We **expect** that things **should** look GOOD, which is like this:]
> `raise "all RESOLVABLE_KIND values **should** be handled in switch";`